auth-foundation/src/main/java/com/okta/authfoundation/credential/Credential.kt

```diff
@@ -69,7 +77,18 @@ class Credential internal constructor(
      */
     val token: Token?
         get() {
-            return _token
+            return state.value.token
+        }
+
+    /**
+     * A [Flow] that emits the current [Token] that's stored and associated with this [Credential].
+     */
+    val tokenFlow: Flow<Token?>
+        get() {
+            return state.transformWhile {
+                emit(it.token)
+                it !is CredentialState.Deleted
```

Comment:
this will emit a final null value before closing the `Flow`, I can rework this a bit if we don't want that final null emitted.

Reply:
at least that is my understanding of how `transformWhile` works, this is my first time using it.